lib/Weight/MockWeight.hpp

```diff
@@ -0,0 +1,31 @@
+#include "IWeight.hpp"
+
+/**
+ * @brief Mock weight class that inherits from the IWeight interface class.
+ * Used for unit tests so that actual hardware isn't require to test the ZotBins
+ * system.
+ */
+
+class MockWeight : IWeight
```

Comment:
Mark the class as `final`, so classes can't inherit from `MockWeight` and make it a public inheritance.